pkg/apis/numaflow/v1alpha1/vertex_types.go

```diff
@@ -343,13 +349,61 @@ func (v Vertex) GetPodSpec(req GetVertexPodSpecReq) (*corev1.PodSpec, error) {
 	}
 
 	if v.IsASource() && v.Spec.Source.Serving != nil {
-		servingContainer, err := v.getServingContainer(req)
+		servingSource := v.Spec.Source.Serving
+		servingSourceBytes, err := json.Marshal(servingSource)
+		if err != nil {
+			return nil, errors.New("failed to marshal serving source spec")
+		}
+		encodedServingSourceSpec := base64.StdEncoding.EncodeToString(servingSourceBytes)
+
+		// Create a SimplifiedPipelineSpec and populate it with the vertex names and edges
+		simplifiedPipelineSpec := PipelineSpec{
+			Vertices: req.PipelineSpec.Vertices,
+			Edges:    req.PipelineSpec.Edges,
+		}
+
+		pipelineSpecBytes, err := json.Marshal(simplifiedPipelineSpec)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("failed to marshal pipeline spec, error: %w", err)
+		}
+		encodedPipelineSpec := base64.StdEncoding.EncodeToString(pipelineSpecBytes)
+
+		containers[0].Env = append(
+			containers[0].Env,
+			// set the serving source stream name in the environment because the numa container will be reading from it
+			corev1.EnvVar{Name: EnvServingJetstreamStream, Value: req.ServingSourceStreamName},
+			corev1.EnvVar{Name: EnvServingObject, Value: encodedServingSourceSpec},
+			corev1.EnvVar{Name: EnvServingMinPipelineSpec, Value: encodedPipelineSpec},
+			corev1.EnvVar{Name: EnvServingPort, Value: strconv.Itoa(VertexHTTPSPort)},
+			corev1.EnvVar{
+				Name: EnvServingHostIP,
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						FieldPath: "status.podIP",
+					},
+				},
+			},
+			corev1.EnvVar{
+				Name:  EnvServingStoreTTL,
+				Value: strconv.Itoa(int(servingSource.Store.GetTTL().Seconds())),
+			},
+		)
+
+		// if auth is configured, set the auth token in the environment
+		if servingSource.Auth != nil && servingSource.Auth.Token != nil {
```

Comment:
This should have been mounted as a volume to the container, there's no need to add env vars to it.